core/src/main/java/fr/sncf/osrd/utils/new_graph/Pathfinding.java

```diff
@@ -0,0 +1,179 @@
+package fr.sncf.osrd.utils.new_graph;
+
+import com.google.common.graph.ImmutableNetwork;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.jetbrains.annotations.NotNull;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@SuppressFBWarnings(
+        value = "FE_FLOATING_POINT_EQUALITY",
+        justification = "No arithmetic is done on values where we test for equality, only copies"
+)
+public class Pathfinding<NodeT, EdgeT> {
+
+    /** Pathfinding step */
+    private record Step<EdgeT>(
+            EdgeRange<EdgeT> range, // Next step
+            Step<EdgeT> prev, // Previous step (to construct the result)
+            double totalDistance, // Total distance from the start
+            double weight, // Priority queue weight (could be different from totalDistance to allow for A*)
+            int nReachedSteps // How many steps have been passed
```

Comment:
in this file, step means either a pathfinding step, or a goal. that's confusing, and I'd very much like that we use a convention which does not use a single words for two distinct things. how about calling the targets (the things not in the queue) goals or just target?